File: 프로그래머스/2/388352. 비밀 코드 해독/비밀 코드 해독.js
function solution(n, q, ans) {
    var answer = 0;
    
    function check(a,b,c,d,e) {
        for(let i = 0; i < q.length; i++) {
            const arr = q[i];
            let sum = 0;
            if(arr.includes(a)) sum++;
            if(arr.includes(b)) sum++;
            if(arr.includes(c)) sum++;
            if(arr.includes(d)) sum++;
            if(arr.includes(e)) sum++;
            if(sum != ans[i]) return false;
        }
        
        return true;
    }
    
    for(let a = 1; a <= n-4; a++) {
        for(let b = a+1; b <= n-3; b++) {
            for(let c = b+1; c <= n-2; c++) {
                for(let d = c+1; d <= n-1; d++) {
                    for(let e = d+1; e <= n; e++) {
                        if(check(a,b,c,d,e)) answer++;
                    }
                }
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/2/388352. 비밀 코드 해독/비밀 코드 해독.py
def solution(n, q, ans):
    answer = 0
    
    def check(a,b,c,d,e):
        for i in range(len(q)):
            arr = q[i]
            count = 0
            if a in arr:
                count+=1
            if b in arr:
                count+=1
            if c in arr:
                count+=1
            if d in arr:
                count+=1
            if e in arr:
                count+=1
            if count != ans[i]:
                return False 
        return True
    
    for a in range(1, n-3):
        for b in range(a+1, n-2):
            for c in range(b+1, n-1):
                for d in range(c+1, n):
                    for e in range(d+1, n+1):
                        if check(a,b,c,d,e):
                            answer+=1
    
    return answer